snuba/query/processors/performance_expressions.py

```diff
@@ -1,40 +1,11 @@
-from typing import Optional
-
-from snuba.query.conditions import (
-    binary_condition,
-    BooleanFunctions,
-    ConditionFunctions,
-)
-from snuba.query.dsl import count, countIf, divide, multiply, plus
-from snuba.query.expressions import (
-    Expression,
-    Literal,
-    Column,
-)
+from snuba.clickhouse.columns import ColumnSet, String, UInt
+from snuba.query.validation.signature import Column as ColType
+from snuba.query.processors.custom_function import CustomFunction
 
 
-def apdex(alias: Optional[str], column: Column, satisfied: Literal) -> Expression:
-    tolerated = multiply(satisfied, Literal(None, 4))
-
-    return divide(
-        plus(
-            countIf(binary_condition(None, ConditionFunctions.LTE, column, satisfied)),
-            divide(
-                countIf(
-                    binary_condition(
-                        None,
-                        BooleanFunctions.AND,
-                        binary_condition(
-                            None, ConditionFunctions.GT, column, satisfied,
-                        ),
-                        binary_condition(
-                            None, ConditionFunctions.LTE, column, tolerated,
-                        ),
-                    ),
-                ),
-                Literal(None, 2),
-            ),
-        ),
-        count(),
-        alias,
-    )
+apdex_processor = CustomFunction(
+    ColumnSet([("column", String()), ("satisfied", UInt(32))]),
+    "apdex",
+    [("column", ColType({String})), ("satisfied", ColType({UInt}))],
+    "divide(plus(countIf(lessOrEquals(column, satisfied)), divide(countIf(and(greater(column, satisfied), lessOrEquals(column, multiply(satisfied, 4)))), 2)), count())",
```

Comment:
You can use infix notation now for arithmetic operations (as long as you do not need parentheses since I forgot to ask Riya to retrofit that change in this part of the language). 

`plus(countIf(lessOrEquals(column, satisfied))  countIf(and(greater(column, satisfied), lessOrEquals(column, satisfied * 4))) / 2) / 2`
(mnaybe)